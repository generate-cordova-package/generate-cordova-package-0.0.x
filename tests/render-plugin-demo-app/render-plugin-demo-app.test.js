/* global it, expect */
/* (see <https://github.com/standard/standard/issues/905>) */

const renderPluginDemoApp = require('../../lib/cordova-plugin/demo-app/render')

const demoAppProperties = {
  demoAppName: 'test-plugin-demo-app',
  id: 'cc.plugintestdemo',
  description: 'plugin demo app',
  version: '0.1.3',
  license: 'ISC',
  author: 'Alice',
  email: 'alice@alice.com;',
  jsObjectName: 'TestObject',
  jsFunctionName: 'testFunction',
  // added by `lib/with-copyright-year.js`
  // when used from the CLI:
  copyrightYear: 2023
}

it('renders fs-tree object correctly for Cordova plugin demo app', () => {
  const outputTree = renderPluginDemoApp(demoAppProperties)

  expect(outputTree).toMatchSnapshot()
})
