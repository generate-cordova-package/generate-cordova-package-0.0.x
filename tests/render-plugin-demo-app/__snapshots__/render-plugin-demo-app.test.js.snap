// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders fs-tree object correctly for Cordova plugin demo app 1`] = `
Object {
  "test-plugin-demo-app": Object {
    "README.md": "# test-plugin-demo-app

Copyright 2023-present Alice <alice@alice.com;>
",
    "config.xml": "<widget id=\\"cc.plugintestdemo\\" version=\\"0.1.3\\">
  <name>test-plugin-demo-app</name>
  <description>plugin demo app</description>
</widget>
",
    "package.json": "{
  \\"name\\": \\"test-plugin-demo-app\\",
  \\"version\\": \\"0.1.3\\",
  \\"// copyright\\": \\"Copyright 2023-present Alice <alice@alice.com;>\\",
  \\"description\\": \\"plugin demo app\\",
  \\"license\\": \\"ISC\\",
  \\"author\\": {
    \\"name\\": \\"Alice\\",
    \\"email\\": \\"alice@alice.com;\\"
  }
}
",
    "www": Object {
      "app.js": "// Copyright 2023-present Alice <alice@alice.com;>

document.addEventListener('deviceready', function () {
  document
    .getElementById('status')
    .appendChild(document.createElement('b'))
    .appendChild(document.createTextNode('ready'))
  window.TestObject.testFunction(function () {
    document
      .getElementById('status')
      .appendChild(document.createElement('i'))
      .appendChild(document.createTextNode('pong'))
  })
})
",
      "body.css": "body {
  background-color: #fff;
}
",
      "index.html": "<html>
  <header>
    <link rel=\\"stylesheet\\" href=\\"body.css\\"/>
  </header>
  <body>
    <h1>test-plugin-demo-app</h1>
    <div id=\\"status\\"/>
    <script src=\\"cordova.js\\"></script>
    <script src=\\"app.js\\"></script>
  </body>
</html>
",
    },
  },
}
`;
